src/jit/lowerxarch.cpp

```diff
@@ -2409,14 +2409,24 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
         }
     }
-    else if (intrinsicID == NI_SSE_Shuffle) // TODO - change to all IMM intrinsics
+    else if (numArgs == 3)
     {
-        assert(op1->OperIsList());
-        GenTree* op3 = op1->AsArgList()->Rest()->Rest()->Current();
-
-        if (op3->IsCnsIntOrI())
+        switch (category)
         {
-            MakeSrcContained(node, op3);
+            case HW_Category_IMM:
+            {
+                assert(op1->OperIsList());
+                GenTree* op3 = op1->AsArgList()->Rest()->Rest()->Current();
+
+                if (op3->IsCnsIntOrI())
+                {
+                    MakeSrcContained(node, op3);
+                }
+                break;
+            }
+
+            default:
```

Comment:
Does this need an assert or log or something to catch the cases that aren't `HW_Category_IMM`?

Reply:
We may want to handle those cases eventually, but not today. As of now, there are still several nodes which do support containment but which don't have the required code in lower yet.

Reply:
Since the question was asked, it might be worth a comment here to the effect that the only intrinsics in this category that will be contained are those for which we normally expect the third operand to be a constant that can be directly encoded.